test/dynamo/test_subclasses.py

```diff
@@ -360,6 +360,10 @@ def test_automatic_dynamic(f, inps, dim_dynamic, exp_frame_count, exp_op_count):
                     fake_inp = fake_mode.from_tensor(
                         inp, dynamic_dims=[dim_dynamic for i in range(x.dim())]
                     )
+                    # Clear the cache, so that we can properly cover the first dynamic compile.
+                    # Alternatively, remove this line and lower the frame counts by 1 below, as the
+                    # first recompile is accounted for via caching.
+                    shape_env.source_to_symint_node_cache.clear()
```

Comment:
This looks potentially BC breaking. cc @thiagocrepaldi 

We may need to have `source_to_symint_node_cache` be opt in, with it being enabled by our FakeTensorMode internally only.

Reply:
Well it passes tests, and seems like a good cache to have, so maybe opt-out?

Reply:
On reflection, I think this test is poorly written and probably shouldn't be reusing the fake tensor mode across recompiles (but if it doesn't, I'm not sure what this test is trying to test...)